src/parser/postgresparser.cpp

```diff
@@ -846,35 +846,41 @@ std::unique_ptr<TableRef> PostgresParser::FromTransform(SelectStmt *select_root)
 }
 
 // Postgres.SelectStmt.groupClause -> terrier.GroupByDescription
-std::unique_ptr<GroupByDescription> PostgresParser::GroupByTransform(List *group, Node *having_node) {
+std::unique_ptr<GroupByDescription> PostgresParser::GroupByTransform(ParseResult *parse_result, List *group,
+                                                                     Node *having_node) {
   if (group == nullptr) {
     return nullptr;
   }
 
-  std::vector<std::shared_ptr<AbstractExpression>> columns;
+  std::vector<common::ManagedPointer<AbstractExpression>> columns;
   for (auto cell = group->head; cell != nullptr; cell = cell->next) {
     auto temp = reinterpret_cast<Node *>(cell->data.ptr_value);
-    columns.emplace_back(ExprTransform(temp));
+    auto expr = ExprTransform(parse_result, temp, nullptr);
+    auto expr_ptr = common::ManagedPointer(expr);
+    parse_result->AddExpression(std::move(expr));
+    columns.emplace_back(expr_ptr);
   }
 
   // TODO(WAN): old system says, having clauses not implemented, depends on AExprTransform
-  std::unique_ptr<AbstractExpression> having = nullptr;
+  auto having = common::ManagedPointer<AbstractExpression>(nullptr);
   if (having_node != nullptr) {
-    having = ExprTransform(having_node);
+    auto expr = ExprTransform(parse_result, having_node, nullptr);
+    having = common::ManagedPointer(expr);
+    parse_result->AddExpression(std::move(expr));
   }
 
-  auto result = std::make_unique<GroupByDescription>(std::move(columns), std::move(having));
+  auto result = std::make_unique<GroupByDescription>(std::move(columns), having);
```

Comment:
Same issue as above, what is preventing parse_result from being deallocated before the result is used?

Reply:
The caller receives both the "tree that relies on ParseResult being alive" and "ParseResult".